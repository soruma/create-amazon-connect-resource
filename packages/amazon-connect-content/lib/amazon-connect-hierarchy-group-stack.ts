import { readFileSync } from 'node:fs';
import * as path from 'node:path';
import * as cdk from 'aws-cdk-lib';
import * as connect from 'aws-cdk-lib/aws-connect';
import { Construct } from 'constructs';
import { parse as JSONCParse } from 'jsonc-parser';

interface HierarchyGroup {
  name: string;
  children?: HierarchyGroup[];
}

/**
 * This stack creates user hierarchy groups for Amazon Connect.
 */
interface AmazonConnectHierarchyGroupProps {
  /**
   * Amazon Connect instance ARN
   */
  connectInstanceArn: string;
}

export class AmazonConnectHierarchyGroupStack extends cdk.NestedStack {
  private props: AmazonConnectHierarchyGroupProps;

  constructor(scope: Construct, id: string, props: AmazonConnectHierarchyGroupProps) {
    super(scope, id);

    this.props = props;

    const data = readFileSync(path.join(__dirname, '..', 'config', 'user-hierarchy-groups.jsonc'), {
      encoding: 'utf8',
    });
    const config = JSONCParse(data) as HierarchyGroup[];

    config.forEach((group, idx) => {
      this.createHierarchyGroup(group, undefined, `Root${idx}`);
    });
  }

  private createHierarchyGroup(group: HierarchyGroup, parentGroupArn: string | undefined, uniqueId: string) {
    const groupId = `${uniqueId}_${group.name.replace(/\s+/g, '')}`;

    const hierarchyGroup = new connect.CfnUserHierarchyGroup(this, groupId, {
      instanceArn: this.props.connectInstanceArn,
      name: group.name,
      parentGroupArn: parentGroupArn,
    });

    if (group.children) {
      group.children.forEach((child, idx) => {
        this.createHierarchyGroup(child, hierarchyGroup.attrUserHierarchyGroupArn, `${groupId}_${idx}`);
      });
    }
  }
}
