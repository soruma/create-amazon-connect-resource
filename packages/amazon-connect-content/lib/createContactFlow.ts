import { readFileSync } from 'node:fs';
import * as path from 'node:path';
import { CfnContactFlow, CfnContactFlowModule } from 'aws-cdk-lib/aws-connect';
import { Construct } from 'constructs';
import { parse as JSONCParse } from 'jsonc-parser';

/**
 * Configuration for contact flow
 */
export interface ContactFlowConfig {
  /**
   * Contact flow id
   */
  readonly id: string;

  readonly file: string;
}

/**
 * Configuration for contact flows
 */
export interface ContactFlowsConfig {
  /**
   * Configuration for module contact flows
   */
  readonly modules: ContactFlowConfig[];

  /**
   * Configuration for contact flows
   */
  readonly contactFlows: ContactFlowConfig[];
}

/**
 * Creates contact flow configurations based on the provided configuration
 * @param {Construct} scope - The CDK Construct scope
 * @param {string} connectInstanceArn - The Amazon Connect instance ARN
 * @returns {ContactFlowsConfig} - The parsed contact flow configuration
 */
export const createContactFlow = (scope: Construct, connectInstanceArn: string): ContactFlowsConfig => {
  const data = readFileSync(path.join(__dirname, '..', 'config', 'contact-flows.jsonc'), { encoding: 'utf8' });
  const config = JSONCParse(data) as ContactFlowsConfig;

  // Add modules
  for (const module of config.modules) {
    const contactFlow = readFileSync(path.join(__dirname, '..', 'config', 'modules', module.file), {
      encoding: 'utf8',
    });
    new CfnContactFlowModule(scope, `ContactFlowModule${module.id}`, {
      instanceArn: connectInstanceArn,
      name: contactFlow.Metadata.name,
      description: contactFlow.Metadata.description,
      content: contactFlow,
    });
  }

  // Add contact flow
  for (const contactFlows of config.contactFlows) {
    const contactFlow = readFileSync(path.join(__dirname, '..', 'config', 'modules', contactFlows.file), {
      encoding: 'utf8',
    });
    new CfnContactFlow(scope, `ContactFlowModule${contactFlows.id}`, {
      instanceArn: connectInstanceArn,
      name: contactFlow.Metadata.name,
      description: contactFlow.Metadata.description,
      type: contactFlow.Metadata.type,
      content: contactFlow,
    });
  }

  return config;
};
