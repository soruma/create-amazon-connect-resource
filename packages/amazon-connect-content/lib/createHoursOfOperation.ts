import { readFileSync } from 'node:fs';
import * as path from 'node:path';
import { CfnHoursOfOperation } from 'aws-cdk-lib/aws-connect';
import { Construct } from 'constructs';
import { parse as JSONCParse } from 'jsonc-parser';

/**
 * Configuration for hours of operation
 */
export interface HoursOfOperationConfig {
  /**
   * Hours of operation id
   */
  readonly id: string;

  /**
   * Hours of operation name
   */
  readonly name: string;

  /**
   * Hours of operation description
   */
  readonly description: string;

  /**
   * Hours of operation timezone
   */
  readonly timeZone: string;

  /**
   * Hours of operation configuration
   */
  readonly config: CfnHoursOfOperation.HoursOfOperationConfigProperty[];

  /**
   * Amazon Connect hours of operation ARN
   */
  hoursOfOperationArn: string;
}

/**
 * Creates hours of operation resources based on the provided configuration
 * @param {Construct} scope - The CDK Construct scope
 * @param {string} connectInstanceArn - The Amazon Connect instance ARN
 * @returns {HoursOfOperationConfig[]} An array of hours of operation configurations
 */
export const createHoursOfOperation = (scope: Construct, connectInstanceArn: string): HoursOfOperationConfig[] => {
  const data = readFileSync(path.join(__dirname, '..', 'config', 'hours-of-operations.jsonc'), { encoding: 'utf8' });
  const configs = JSONCParse(data) as HoursOfOperationConfig[];

  for (const config of configs) {
    const hoursOfOperation = new CfnHoursOfOperation(scope, `HoursOfOperation${config.id}`, {
      instanceArn: connectInstanceArn,
      ...config,
    });
    config.hoursOfOperationArn = hoursOfOperation.attrHoursOfOperationArn;
  }

  return configs;
};
