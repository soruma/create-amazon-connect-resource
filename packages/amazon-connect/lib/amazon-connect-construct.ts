import * as connect from 'aws-cdk-lib/aws-connect';
import * as iam from 'aws-cdk-lib/aws-iam';
import * as kms from 'aws-cdk-lib/aws-kms';
import * as s3 from 'aws-cdk-lib/aws-s3';
import { Construct } from 'constructs';

export type IdentityManagementType = 'SAML' | 'CONNECT_MANAGED' | 'EXISTING_DIRECTORY';

interface AmazonConnectConstructProps {
  /**
   * Amazon Connect instance alias
   */
  connectInstanceAlias: string;

  /**
   * Mandatory element which enables inbound calls on new instance.
   */
  inboundCalls: boolean;

  /**
   * Mandatory element which enables outbound calls on new instance.
   */
  outboundCalls: boolean;

  /**
   * Boolean flag which enables CONTACTFLOW_LOGS on an instance.
   */
  contactflowLogs: boolean;

  /**
   * Boolean flag which enables AUTO_RESOLVE_BEST_VOICES on an instance.
   */
  autoResolveBestVoices: boolean;

  /**
   * The identity management type.
   */
  identityManagementType: IdentityManagementType;

  /**
   * The identifier for the directory.
   */
  directoryId?: string;

  /**
   * Amazon connect data storage
   */
  dataStorageBucket?: s3.Bucket;
}

export class AmazonConnectConstruct extends Construct {
  private readonly props: AmazonConnectConstructProps;
  readonly connectInstance: connect.CfnInstance;

  constructor(scope: Construct, id: string, props: AmazonConnectConstructProps) {
    super(scope, id);

    this.props = props;

    this.connectInstance = new connect.CfnInstance(this, id, {
      attributes: {
        inboundCalls: this.props.inboundCalls,
        outboundCalls: this.props.outboundCalls,
        contactflowLogs: this.props.contactflowLogs,
        autoResolveBestVoices: this.props.autoResolveBestVoices,
      },
      identityManagementType: this.props.identityManagementType,
      instanceAlias: this.props.connectInstanceAlias,
      directoryId: this.props.directoryId,
    });

    if (this.props.dataStorageBucket !== undefined) {
      this.props.dataStorageBucket.grantWrite(new iam.ServicePrincipal('connect.amazonaws.com'));

      this.createInstanceStorageConfig();
    }
  }

  createInstanceStorageConfig() {
    if (this.props.dataStorageBucket === undefined) return;

    const encriptionKey = kms.Key.fromLookup(this, 'EncryptKeyLookup', {
      aliasName: 'alias/aws/connect',
    });

    new connect.CfnInstanceStorageConfig(this, 'CallRecordingsStorageConfig', {
      instanceArn: this.connectInstance.attrArn,
      resourceType: 'CALL_RECORDINGS',
      storageType: 'S3',
      s3Config: {
        bucketName: this.props.dataStorageBucket.bucketName,
        bucketPrefix: 'CallRecordings',
        encryptionConfig: {
          encryptionType: 'KMS',
          keyId: encriptionKey.keyArn,
        },
      },
    });

    new connect.CfnInstanceStorageConfig(this, 'ChatTranscriptsStorageConfig', {
      instanceArn: this.connectInstance.attrArn,
      resourceType: 'CHAT_TRANSCRIPTS',
      storageType: 'S3',
      s3Config: {
        bucketName: this.props.dataStorageBucket.bucketName,
        bucketPrefix: 'ChatTranscripts',
        encryptionConfig: {
          encryptionType: 'KMS',
          keyId: encriptionKey.keyArn,
        },
      },
    });
  }
}
