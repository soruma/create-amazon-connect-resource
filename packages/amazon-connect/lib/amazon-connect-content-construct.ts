import { readFileSync } from 'node:fs';
import * as path from 'node:path';
import * as connect from 'aws-cdk-lib/aws-connect';
import { Construct } from 'constructs';

import { AmazonConnectContentHierarchyGroupStack } from './amazon-connect-content-hierarchy-group-stack';

interface AmazonConnectContentConstructProps {
  /**
   * Amazon Connect instance ARN
   */
  connectInstanceArn: string;

  /**
   * Create sample Amazon Connect organization hierarchies
   */
  createHierarchy: boolean;

  /**
   * business hours time zone
   */
  businessHoursTimeZone: string;
}

export class AmazonConnectContentConstruct extends Construct {
  private props: AmazonConnectContentConstructProps;

  constructor(scope: Construct, id: string, props: AmazonConnectContentConstructProps) {
    super(scope, id);

    this.props = props;

    this.createBusinessHours();

    if (this.props.createHierarchy) {
      this.createHierarchy();
    }
  }

  createBusinessHours(): connect.CfnHoursOfOperation {
    const data = readFileSync(path.join(__dirname, '..', 'config', 'business_hours.json'), { encoding: 'utf8' });
    const config = JSON.parse(data) as connect.CfnHoursOfOperation.HoursOfOperationConfigProperty[];

    return new connect.CfnHoursOfOperation(this, 'HoursOfOperation', {
      instanceArn: this.props.connectInstanceArn,
      name: 'Business Hour',
      config,
      timeZone: this.props.businessHoursTimeZone,
    });
  }

  createHierarchy() {
    new connect.CfnUserHierarchyStructure(this, 'DepartmentUserHierarchyStructure', {
      instanceArn: this.props.connectInstanceArn,
      userHierarchyStructure: {
        levelOne: {
          name: 'Department',
        },
        levelTwo: {
          name: 'Team',
        },
      },
    });

    new AmazonConnectContentHierarchyGroupStack(this, 'AmazonConnectContentHierarchyGroupStack', {
      connectInstanceArn: this.props.connectInstanceArn,
    });
  }
}
