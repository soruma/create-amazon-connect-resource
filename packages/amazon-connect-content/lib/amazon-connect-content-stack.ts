import { readFileSync } from 'node:fs';
import * as path from 'node:path';
import * as cdk from 'aws-cdk-lib';
import * as connect from 'aws-cdk-lib/aws-connect';
import { Construct } from 'constructs';

import { AmazonConnectHierarchyGroupStack } from './amazon-connect-hierarchy-group-stack';

interface AmazonConnectContentStackProps extends cdk.StackProps {
  /**
   * Amazon Connect instance ARN
   */
  connectInstanceArn: string;

  /**
   * Create sample Amazon Connect organization hierarchies
   */
  createHierarchy: boolean;

  /**
   * business hours time zone
   */
  businessHoursTimeZone: string;
}

export class AmazonConnectContentStack extends cdk.Stack {
  private readonly props: AmazonConnectContentStackProps;

  constructor(scope: Construct, id: string, props: AmazonConnectContentStackProps) {
    super(scope, id, props);

    this.props = props;

    this.createBusinessHours();

    if (this.props.createHierarchy) {
      this.createHierarchy();
    }
  }

  createBusinessHours(): connect.CfnHoursOfOperation {
    const data = readFileSync(path.join(__dirname, '..', 'config', 'business_hours.json'), { encoding: 'utf8' });
    const config = JSON.parse(data) as connect.CfnHoursOfOperation.HoursOfOperationConfigProperty[];

    return new connect.CfnHoursOfOperation(this, 'HoursOfOperation', {
      instanceArn: this.props.connectInstanceArn,
      name: 'Business Hour',
      config,
      timeZone: this.props.businessHoursTimeZone,
    });
  }

  createHierarchy() {
    new connect.CfnUserHierarchyStructure(this, 'DepartmentUserHierarchyStructure', {
      instanceArn: this.props.connectInstanceArn,
      userHierarchyStructure: {
        levelOne: {
          name: 'Department',
        },
        levelTwo: {
          name: 'Team',
        },
      },
    });

    new AmazonConnectHierarchyGroupStack(this, 'AmazonConnectHierarchyGroupStack', {
      connectInstanceArn: this.props.connectInstanceArn,
    });
  }
}
