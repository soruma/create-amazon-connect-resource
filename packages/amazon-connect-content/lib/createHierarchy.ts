import { readFileSync } from 'node:fs';
import * as path from 'node:path';
import { CfnUserHierarchyStructure } from 'aws-cdk-lib/aws-connect';
import { Construct } from 'constructs';
import { parse as JSONCParse } from 'jsonc-parser';

import { AmazonConnectHierarchyGroupStack } from './amazon-connect-hierarchy-group-stack';

/**
 * Creates a hierarchy group in Amazon Connect.
 * @param {Construct} scope - The CDK Construct scope
 * @param {string} connectInstanceArn - The Amazon Connect instance ARN
 */
export const createHierarchy = (scope: Construct, connectInstanceArn: string) => {
  const data = readFileSync(path.join(__dirname, '..', 'config', 'hierarchy-structure.jsonc'), { encoding: 'utf8' });
  const userHierarchyStructure = JSONCParse(data) as CfnUserHierarchyStructure.UserHierarchyStructureProperty;

  new CfnUserHierarchyStructure(scope, 'DepartmentUserHierarchyStructure', {
    instanceArn: connectInstanceArn,
    userHierarchyStructure,
  });

  new AmazonConnectHierarchyGroupStack(scope, 'AmazonConnectHierarchyGroupStack', {
    connectInstanceArn: connectInstanceArn,
  });
};
