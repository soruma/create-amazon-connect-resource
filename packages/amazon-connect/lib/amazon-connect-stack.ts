import * as cdk from 'aws-cdk-lib';
import * as s3 from 'aws-cdk-lib/aws-s3';
import { Construct } from 'constructs';

import { AmazonConnectConstruct, IdentityManagementType } from './amazon-connect-construct';

interface AmazonConnectStackProps extends cdk.StackProps {
  /**
   * Amazon Connect instance alias
   */
  connectInstanceAlias: string;

  /**
   * Whether to store call and chat records in storage
   */
  createDataStorageBucket: boolean;

  /**
   * Mandatory element which enables inbound calls on new instance.
   */
  inboundCalls: boolean;

  /**
   * Mandatory element which enables outbound calls on new instance.
   */
  outboundCalls: boolean;

  /**
   * Boolean flag which enables CONTACTFLOW_LOGS on an instance.
   */
  contactflowLogs: boolean;

  /**
   * Boolean flag which enables AUTO_RESOLVE_BEST_VOICES on an instance.
   */
  autoResolveBestVoices: boolean;

  /**
   * The identity management type.
   */
  identityManagementType: IdentityManagementType;

  /**
   * The identifier for the directory.
   */
  directoryId?: string;
}

export class AmazonConnectStack extends cdk.Stack {
  private props: AmazonConnectStackProps;

  constructor(scope: Construct, id: string, props: AmazonConnectStackProps) {
    super(scope, id, props);

    this.props = props;

    let dataStorageBucket: s3.Bucket | undefined;

    if (this.props.createDataStorageBucket) {
      dataStorageBucket = new s3.Bucket(this, 'DataStorageBucket', {
        bucketName: `amazon-connect-${this.props.connectInstanceAlias}`,
        encryption: s3.BucketEncryption.S3_MANAGED,
      });
    }

    const amazonConnect = new AmazonConnectConstruct(this, 'AmazonConnectConstruct', {
      connectInstanceAlias: this.props.connectInstanceAlias,
      inboundCalls: this.props.inboundCalls,
      outboundCalls: this.props.outboundCalls,
      contactflowLogs: this.props.contactflowLogs,
      autoResolveBestVoices: this.props.autoResolveBestVoices,
      identityManagementType: this.props.identityManagementType,
      directoryId: this.props.directoryId,
      dataStorageBucket,
    });

    new cdk.CfnOutput(this, 'AmazonConnectInstanceArn', {
      exportName: `AmazonConnectInstanceArn-${this.props.connectInstanceAlias}`,
      value: amazonConnect.connectInstance.attrArn,
    });
  }
}
