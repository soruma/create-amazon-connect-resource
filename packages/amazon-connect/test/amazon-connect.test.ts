import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { test as baseTest, describe, expect, it } from 'vitest';

import { AmazonConnectStack } from '../lib/amazon-connect-stack';

const test = baseTest.extend<{
  app: cdk.App;
}>({
  // biome-ignore lint: lint/correctness/noEmptyPattern
  app: ({}, use) => use(new cdk.App()),
});

describe('AmazonConnectStack', () => {
  describe('Matches the snapshot', () => {
    test('Create data storage', ({ app }) => {
      const stack = new AmazonConnectStack(app, 'AmazonConnectStack-test', {
        env: {
          account: '000000000000',
          region: 'us-east-1',
        },
        connectInstanceAlias: 'test',
        attributes: {
          inboundCalls: true,
          outboundCalls: true,
          contactflowLogs: true,
          autoResolveBestVoices: true,
        },
        identityManagementType: 'CONNECT_MANAGED',
        createDataStorageBucket: true,
      });

      const template = Template.fromStack(stack);
      expect(template.toJSON()).toMatchSnapshot();
    });

    it('Not create data storage', () => {
      const app = new cdk.App();

      const stack = new AmazonConnectStack(app, 'AmazonConnectStack-test', {
        env: {
          account: '000000000000',
          region: 'us-east-1',
        },
        connectInstanceAlias: 'test',
        attributes: {
          inboundCalls: true,
          outboundCalls: true,
          contactflowLogs: true,
          autoResolveBestVoices: true,
        },
        identityManagementType: 'CONNECT_MANAGED',
        createDataStorageBucket: false,
      });

      const template = Template.fromStack(stack);
      expect(template.toJSON()).toMatchSnapshot();
    });

    it('Identity management is Directory service', () => {
      const app = new cdk.App();

      const stack = new AmazonConnectStack(app, 'AmazonConnectStack-test', {
        env: {
          account: '000000000000',
          region: 'us-east-1',
        },
        connectInstanceAlias: 'test',
        attributes: {
          inboundCalls: true,
          outboundCalls: true,
          contactflowLogs: true,
          autoResolveBestVoices: true,
        },
        identityManagementType: 'EXISTING_DIRECTORY',
        directoryId: 'd-0000000000',
        createDataStorageBucket: true,
      });

      const template = Template.fromStack(stack);
      expect(template.toJSON()).toMatchSnapshot();
    });
  });
});
