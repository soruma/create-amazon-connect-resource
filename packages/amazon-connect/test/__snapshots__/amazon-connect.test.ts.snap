// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AmazonConnectStack > Matches the snapshot > Create data storage 1`] = `
{
  "Outputs": {
    "AmazonConnectInstanceArn": {
      "Export": {
        "Name": "AmazonConnectInstanceArn-test",
      },
      "Value": {
        "Fn::GetAtt": [
          "AmazonConnectConstruct1C9DFEFB",
          "Arn",
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "AmazonConnectConstruct1C9DFEFB": {
      "Properties": {
        "Attributes": {
          "AutoResolveBestVoices": true,
          "ContactflowLogs": true,
          "InboundCalls": true,
          "OutboundCalls": true,
        },
        "IdentityManagementType": "CONNECT_MANAGED",
        "InstanceAlias": "test",
      },
      "Type": "AWS::Connect::Instance",
    },
    "AmazonConnectConstructCallRecordingsStorageConfig2138FB32": {
      "Properties": {
        "InstanceArn": {
          "Fn::GetAtt": [
            "AmazonConnectConstruct1C9DFEFB",
            "Arn",
          ],
        },
        "ResourceType": "CALL_RECORDINGS",
        "S3Config": {
          "BucketName": {
            "Ref": "DataStorageBucket1A195487",
          },
          "BucketPrefix": "CallRecordings",
          "EncryptionConfig": {
            "EncryptionType": "KMS",
            "KeyId": {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition",
                  },
                  ":kms:us-east-1:000000000000:key/1234abcd-12ab-34cd-56ef-1234567890ab",
                ],
              ],
            },
          },
        },
        "StorageType": "S3",
      },
      "Type": "AWS::Connect::InstanceStorageConfig",
    },
    "AmazonConnectConstructChatTranscriptsStorageConfig0CCF0AA9": {
      "Properties": {
        "InstanceArn": {
          "Fn::GetAtt": [
            "AmazonConnectConstruct1C9DFEFB",
            "Arn",
          ],
        },
        "ResourceType": "CHAT_TRANSCRIPTS",
        "S3Config": {
          "BucketName": {
            "Ref": "DataStorageBucket1A195487",
          },
          "BucketPrefix": "ChatTranscripts",
          "EncryptionConfig": {
            "EncryptionType": "KMS",
            "KeyId": {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition",
                  },
                  ":kms:us-east-1:000000000000:key/1234abcd-12ab-34cd-56ef-1234567890ab",
                ],
              ],
            },
          },
        },
        "StorageType": "S3",
      },
      "Type": "AWS::Connect::InstanceStorageConfig",
    },
    "DataStorageBucket1A195487": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "BucketName": "amazon-connect-test",
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "DataStorageBucketPolicyAD94CA11": {
      "Properties": {
        "Bucket": {
          "Ref": "DataStorageBucket1A195487",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Principal": {
                "Service": "connect.amazonaws.com",
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "DataStorageBucket1A195487",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "DataStorageBucket1A195487",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`AmazonConnectStack > Matches the snapshot > Identity management is Directory service 1`] = `
{
  "Outputs": {
    "AmazonConnectInstanceArn": {
      "Export": {
        "Name": "AmazonConnectInstanceArn-test",
      },
      "Value": {
        "Fn::GetAtt": [
          "AmazonConnectConstruct1C9DFEFB",
          "Arn",
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "AmazonConnectConstruct1C9DFEFB": {
      "Properties": {
        "Attributes": {
          "AutoResolveBestVoices": true,
          "ContactflowLogs": true,
          "InboundCalls": true,
          "OutboundCalls": true,
        },
        "DirectoryId": "d-0000000000",
        "IdentityManagementType": "EXISTING_DIRECTORY",
        "InstanceAlias": "test",
      },
      "Type": "AWS::Connect::Instance",
    },
    "AmazonConnectConstructCallRecordingsStorageConfig2138FB32": {
      "Properties": {
        "InstanceArn": {
          "Fn::GetAtt": [
            "AmazonConnectConstruct1C9DFEFB",
            "Arn",
          ],
        },
        "ResourceType": "CALL_RECORDINGS",
        "S3Config": {
          "BucketName": {
            "Ref": "DataStorageBucket1A195487",
          },
          "BucketPrefix": "CallRecordings",
          "EncryptionConfig": {
            "EncryptionType": "KMS",
            "KeyId": {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition",
                  },
                  ":kms:us-east-1:000000000000:key/1234abcd-12ab-34cd-56ef-1234567890ab",
                ],
              ],
            },
          },
        },
        "StorageType": "S3",
      },
      "Type": "AWS::Connect::InstanceStorageConfig",
    },
    "AmazonConnectConstructChatTranscriptsStorageConfig0CCF0AA9": {
      "Properties": {
        "InstanceArn": {
          "Fn::GetAtt": [
            "AmazonConnectConstruct1C9DFEFB",
            "Arn",
          ],
        },
        "ResourceType": "CHAT_TRANSCRIPTS",
        "S3Config": {
          "BucketName": {
            "Ref": "DataStorageBucket1A195487",
          },
          "BucketPrefix": "ChatTranscripts",
          "EncryptionConfig": {
            "EncryptionType": "KMS",
            "KeyId": {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition",
                  },
                  ":kms:us-east-1:000000000000:key/1234abcd-12ab-34cd-56ef-1234567890ab",
                ],
              ],
            },
          },
        },
        "StorageType": "S3",
      },
      "Type": "AWS::Connect::InstanceStorageConfig",
    },
    "DataStorageBucket1A195487": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "BucketName": "amazon-connect-test",
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "DataStorageBucketPolicyAD94CA11": {
      "Properties": {
        "Bucket": {
          "Ref": "DataStorageBucket1A195487",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Principal": {
                "Service": "connect.amazonaws.com",
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "DataStorageBucket1A195487",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "DataStorageBucket1A195487",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`AmazonConnectStack > Matches the snapshot > Not create data storage 1`] = `
{
  "Outputs": {
    "AmazonConnectInstanceArn": {
      "Export": {
        "Name": "AmazonConnectInstanceArn-test",
      },
      "Value": {
        "Fn::GetAtt": [
          "AmazonConnectConstruct1C9DFEFB",
          "Arn",
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "AmazonConnectConstruct1C9DFEFB": {
      "Properties": {
        "Attributes": {
          "AutoResolveBestVoices": true,
          "ContactflowLogs": true,
          "InboundCalls": true,
          "OutboundCalls": true,
        },
        "IdentityManagementType": "CONNECT_MANAGED",
        "InstanceAlias": "test",
      },
      "Type": "AWS::Connect::Instance",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
